client/data/viewers/use-viewers-query.js

```diff
@@ -12,27 +12,22 @@ import wpcom from 'calypso/lib/wp';
 const extractPages = ( pages = [] ) => pages.flatMap( ( page ) => page.viewers );
 const compareUnique = ( a, b ) => a.ID === b.ID;
 
-const DEFAULT_NUMBER = 100;
 const defaults = {
-	page: 1,
 	number: 100,
 };
 
-const useViewers = ( fetchOptions = {}, queryOptions = {} ) => {
-	const { siteId } = fetchOptions;
-
+const useViewersQuery = ( siteId, fetchOptions = {}, queryOptions = {} ) => {
 	return useInfiniteQuery(
 		[ 'viewers', siteId ],
-		async ( { pageParam = 1 } ) => {
-			const res = await wpcom
-				.undocumented()
-				.site( siteId )
-				.getViewers( { ...defaults, ...fetchOptions, page: pageParam } );
-			return res;
-		},
+		( { pageParam = 1 } ) =>
+			wpcom.req.get( `/sites/${ siteId }/viewers`, {
+				...defaults,
+				...fetchOptions,
+				page: pageParam,
+			} ),
 		{
 			getNextPageParam: ( lastPage, allPages ) => {
-				if ( lastPage.found <= allPages.length * DEFAULT_NUMBER ) {
+				if ( lastPage.found <= allPages.length * defaults.number ) {
```

Comment:
`number` can be overridden in `fetchOptions`, so this condition should look at `fetchOption.number`. Just like the `useUsersQuery()` hook already does.